### PR TITLE
622 GitHub abstraction

### PR DESCRIPTION
**Pull request by nutrina**
This fixes:

#622 
#632

### CODE REVIEW

**Review by aminah-io — COMMENTED** [review]
Left a few suggestions in there for ya!

File: platforms/src/Gitcoin/App-Bindings.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  redirectUri: string = null;
+
+  constructor(options: PlatformOptions = {}) {
+    this.clientId = options.clientId as string;
+    this.redirectUri = options.redirectUri as string;
+  }
+
+  async getOAuthUrl(state: string): Promise<string> {
+    const githubUrl = `https://github.com/login/oauth/authorize?client_id=${this.clientId}&redirect_uri=${this.redirectUri}&state=${state}`;
+    console.log("geri GithubPlatform getOAuthUrl ", githubUrl);
```
[the remark this answers is not in the document]
Comment (nutrina) [reply platforms/src/Gitcoin/App-Bindings.ts]: This should be resolved now

File: platforms/index.ts [hunk 2]
```diff
@@ -9,3 +9,21 @@ export {
...
+  FiveOrMoreGithubRepos,
+} from "./src/Github/Providers";
+
+
+export * as types from "./src/types";
+export * as Twitter from "./src/Twitter";
+export * as Github from "./src/Github";
+export * as Gitcoin from "./src/Gitcoin";
+
+
```
[the remark this answers is not in the document]
Comment (nutrina) [reply platforms/index.ts]: This should be resolved now

File: platforms/index.ts [hunk 3]
```diff
@@ -9,3 +9,21 @@ export {
...
+export {
+  GithubProvider,
+  ForkedGithubRepoProvider,
+  StarredGithubRepoProvider,
+  TenOrMoreGithubFollowers,
+  FiftyOrMoreGithubFollowers,
+  FiveOrMoreGithubRepos,
+} from "./src/Github/Providers";
+
```
[the remark this answers is not in the document]
Comment (nutrina) [reply platforms/index.ts]: This should be resolved now

File: iam/src/index.ts [hunk 4]
```diff
@@ -49,18 +49,20 @@ import {
...
+  FiftyOrMoreGithubFollowers,
+} from "@gitcoin/passport-platforms";
 import { PohProvider } from "./providers/poh";
 import { POAPProvider } from "./providers/poap";
 import { FacebookProvider } from "./providers/facebook";
 import { FacebookFriendsProvider } from "./providers/facebookFriends";
 import { FacebookProfilePictureProvider } from "./providers/facebookProfilePicture";
 import { BrightIdProvider } from "./providers/brightid";
-import { GithubProvider } from "./providers/github";
 import { FiveOrMoreGithubRepos } from "./providers/fiveOrMoreGithubRepos";
```
[the remark this answers is not in the document]
Comment (nutrina) [reply iam/src/index.ts]: This should be fixed now.

**Review by aminah-io — APPROVED** [review]
LGTM 🚀

**Pull request merged**